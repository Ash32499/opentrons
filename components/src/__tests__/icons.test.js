// icon components tests
import React from 'react'
import Renderer from 'react-test-renderer'

import {
  Icon,
  ALERT,
  BACK,
  REFRESH,
  SPINNER,
  USB,
  WIFI,
  FLASK,
  CHECKED,
  UNCHECKED,
  EXPAND,
  COLLAPSE,
  PROTOCOL,
  COG,
  CONNECT
} from '..'

describe('icons', () => {
  test('alert icon renders correctly', () => {
    const tree = Renderer.create(
      <Icon name={ALERT} className='foo' />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('back icon renders correctly', () => {
    const tree = Renderer.create(
      <Icon name={BACK} className='foo' />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('refresh icon renders correctly', () => {
    const tree = Renderer.create(
      <Icon name={REFRESH} className='foo' />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('spinner icon renders correctly', () => {
    const tree = Renderer.create(
      <Icon name={SPINNER} className='foo' />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('usb icon renders correctly', () => {
    const tree = Renderer.create(
      <Icon name={USB} className='foo' />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('wifi icon renders correctly', () => {
    const tree = Renderer.create(
      <Icon name={WIFI} className='foo' />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('spin option adds spin class', () => {
    const icon = Renderer.create(
      <Icon name={SPINNER} spin />
    ).root.findByType('svg')

    expect(icon.props.className).toMatch(/\bspin\b/)
  })

  test('flask icon renders correctly', () => {
    const tree = Renderer.create(
      <Icon name={FLASK} className='foo' />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('checked icon renders correctly', () => {
    const tree = Renderer.create(
      <Icon name={CHECKED} className='foo' />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('unchecked icon renders correctly', () => {
    const tree = Renderer.create(
      <Icon name={UNCHECKED} className='foo' />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('expand caret icon renders correctly', () => {
    const tree = Renderer.create(
      <Icon name={EXPAND} className='foo' />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('collapse caret icon renders correctly', () => {
    const tree = Renderer.create(
      <Icon name={COLLAPSE} className='foo' />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('protocol file icon renders correctly', () => {
    const tree = Renderer.create(
      <Icon name={PROTOCOL} className='foo' />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('setup cog icon renders correctly', () => {
    const tree = Renderer.create(
      <Icon name={COG} className='foo' />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('connect icon renders correctly', () => {
    const tree = Renderer.create(
      <Icon name={CONNECT} className='foo' />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
