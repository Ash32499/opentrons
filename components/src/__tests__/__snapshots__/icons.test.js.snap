// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icons alert icon renders correctly 1`] = `
<svg
  aria-hidden="true"
  className="foo"
  fill="currentColor"
  height={undefined}
  version="1.1"
  viewBox="0 0 24 24"
  width={undefined}
  x={undefined}
  y={undefined}
>
  <path
    d="M13.2 13.2h-2.4V6h2.4v7.2zm0 4.8h-2.4v-2.4h2.4V18zM12 0C5.373 0 0 5.373 0 12A12 12 0 1 0 12 0z"
    fillRule="evenodd"
  />
</svg>
`;

exports[`icons back icon renders correctly 1`] = `
<svg
  aria-hidden="true"
  className="foo"
  fill="currentColor"
  height={undefined}
  version="1.1"
  viewBox="0 0 24 24"
  width={undefined}
  x={undefined}
  y={undefined}
>
  <path
    d="M18 11.242v1.516H8.91l4.166 4.166L12 18l-6-6 6-6 1.076 1.076-4.167 4.166z"
    fillRule="evenodd"
  />
</svg>
`;

exports[`icons flask icon renders correctly 1`] = `
<svg
  aria-hidden="true"
  className="foo"
  fill="currentColor"
  height={undefined}
  version="1.1"
  viewBox="0 0 24 24"
  width={undefined}
  x={undefined}
  y={undefined}
>
  <path
    d="M2.44444444,20.4 C2.44444444,21.0627417 2.99165197,21.6 3.66666667,21.6 L18.3333333,21.6 C19.008348,21.6 19.5555556,21.0627417 19.5555556,20.4 C19.5555556,20.148 19.47,19.908 19.3355556,19.716 L12.2222222,7.62 L12.2222222,2.4 L9.77777778,2.4 L9.77777778,7.62 L2.66444444,19.716 C2.53,19.908 2.44444444,20.148 2.44444444,20.4 Z M3.66666667,24 C1.64162258,24 0,22.3882251 0,20.4 C0,19.68 0.22,19.008 0.611111111,18.444 L7.33333333,6.972 L7.33333333,4.8 C6.65831864,4.8 6.11111111,4.2627417 6.11111111,3.6 L6.11111111,2.4 C6.11111111,1.0745166 7.20552617,0 8.55555556,0 L13.4444444,0 C14.7944738,0 15.8888889,1.0745166 15.8888889,2.4 L15.8888889,3.6 C15.8888889,4.2627417 15.3416814,4.8 14.6666667,4.8 L14.6666667,6.972 L21.3888889,18.444 C21.78,19.008 22,19.68 22,20.4 C22,22.3882251 20.3583774,24 18.3333333,24 L3.66666667,24 Z M12.2222222,16.8 L13.86,15.192 L16.2188889,19.2 L5.78111111,19.2 L9.03222222,13.668 L12.2222222,16.8 Z M11.6111111,12 C11.9486185,12 12.2222222,12.2686292 12.2222222,12.6 C12.2222222,12.9313708 11.9486185,13.2 11.6111111,13.2 C11.2736038,13.2 11,12.9313708 11,12.6 C11,12.2686292 11.2736038,12 11.6111111,12 Z"
    fillRule="evenodd"
  />
</svg>
`;

exports[`icons refresh icon renders correctly 1`] = `
<svg
  aria-hidden="true"
  className="foo"
  fill="currentColor"
  height={undefined}
  version="1.1"
  viewBox="0 0 200 200"
  width={undefined}
  x={undefined}
  y={undefined}
>
  <path
    d="M121.9,31.3L129,4.6l56.7,56.7l-77.5,20.8c0,0,7.7-28.6,7.7-28.6c-5.7-1.8-11.7-2.7-17.9-2.7 c-33.5,0-60.8,27.3-60.8,60.8c0,33.5,27.3,60.8,60.8,60.8c26.5,0,50.5-17.7,58.1-43.1l22,6.7c-5.1,16.8-15.7,32-29.8,42.6 c-14.6,11-32,16.8-50.3,16.8c-46.2,0-83.8-37.6-83.8-83.8S51.8,27.9,98,27.9C106.1,27.9,114.1,29.1,121.9,31.3"
    fillRule="evenodd"
  />
</svg>
`;

exports[`icons spinner icon renders correctly 1`] = `
<svg
  aria-hidden="true"
  className="foo"
  fill="currentColor"
  height={undefined}
  version="1.1"
  viewBox="0 0 512 512"
  width={undefined}
  x={undefined}
  y={undefined}
>
  <path
    d="M304 48c0 26.51-21.49 48-48 48s-48-21.49-48-48 21.49-48 48-48 48 21.49 48 48zm-48 368c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm208-208c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zM96 256c0-26.51-21.49-48-48-48S0 229.49 0 256s21.49 48 48 48 48-21.49 48-48zm12.922 99.078c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.491-48-48-48zm294.156 0c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.49-48-48-48zM108.922 60.922c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.491-48-48-48z"
    fillRule="evenodd"
  />
</svg>
`;

exports[`icons usb icon renders correctly 1`] = `
<svg
  aria-hidden="true"
  className="foo"
  fill="currentColor"
  height={undefined}
  version="1.1"
  viewBox="0 0 24 24"
  width={undefined}
  x={undefined}
  y={undefined}
>
  <path
    d="M15.347 6.792v4.529h1.132v2.264h-3.396V4.528h2.264L11.951 0 8.555 4.528h2.264v9.057H7.423v-2.343c.792-.42 1.358-1.223 1.358-2.185 0-1.37-1.12-2.491-2.49-2.491S3.8 7.686 3.8 9.056c0 .963.566 1.767 1.358 2.186v2.343a2.256 2.256 0 0 0 2.265 2.264h3.396v3.453a2.492 2.492 0 0 0-1.359 2.207 2.49 2.49 0 0 0 4.982 0c0-.962-.555-1.788-1.359-2.207v-3.453h3.396a2.256 2.256 0 0 0 2.264-2.264V11.32h1.132V6.792h-4.528z"
    fillRule="evenodd"
  />
</svg>
`;

exports[`icons wifi icon renders correctly 1`] = `
<svg
  aria-hidden="true"
  className="foo"
  fill="currentColor"
  height={undefined}
  version="1.1"
  viewBox="0 0 24 24"
  width={undefined}
  x={undefined}
  y={undefined}
>
  <path
    d="M0 9.414l2.182 2.182c5.422-5.422 14.214-5.422 19.636 0L24 9.414c-6.622-6.622-17.367-6.622-24 0zm8.727 8.727L12 21.414l3.273-3.273a4.622 4.622 0 0 0-6.546 0zm-4.363-4.363l2.181 2.181a7.717 7.717 0 0 1 10.91 0l2.181-2.181c-4.21-4.211-11.05-4.211-15.272 0z"
    fillRule="evenodd"
  />
</svg>
`;
